src/amo/components/AddonRecommendations/index.js

```diff
@@ -0,0 +1,184 @@
+|};
+
+
+export class AddonRecommendationsBase extends React.Component<Props> {
+  static defaultProps = {
+    cookie: defaultCookie,
+    randomizer: Math.random,
+    recommendations: null,
+    tracking: defaultTracking,
+  };
+
+  componentDidMount() {
+    const { cookie, randomizer, addon, recommendations } = this.props;
+
+    // Set a cohort for the experiment.
+    this.cohort = cookie.load(TAAR_COHORT_COOKIE_NAME);
+    if (this.cohort === undefined) {
+      this.cohort = randomizer() > 0.5 ?
+        TAAR_COHORT_INCLUDED : TAAR_COHORT_EXCLUDED;
+      cookie.save(TAAR_COHORT_COOKIE_NAME, this.cohort, { path: '/' });
+    }
+
+    if (addon && !recommendations) {
+      this.dispatchFetchRecommendations({
+        guid: addon.guid,
+        recommended: this.cohort === TAAR_COHORT_INCLUDED,
+      });
+    }
+  }
+
+  componentWillReceiveProps({
+    addon: newAddon,
+    recommendations: newRecommendations,
+  }: Props) {
+    const {
+      addon: oldAddon,
+      recommendations: oldRecommendations,
+      tracking,
+    } = this.props;
+
+    // Fetch recommendations when the add-on changes.
+    if (newAddon && oldAddon !== newAddon) {
+      this.dispatchFetchRecommendations({
+        guid: newAddon.guid,
+        recommended: this.cohort === TAAR_COHORT_INCLUDED,
+      });
+    }
+
+    // Send the GA ping when recommendations are loaded.
+    if (newRecommendations && oldRecommendations !== newRecommendations) {
+      const { fallbackReason, loading, outcome } = newRecommendations;
+
+      if (loading) {
+        return;
+      }
+
+      invariant(newAddon, 'newAddon is required');
+      invariant(outcome, 'outcome is required');
+
+      // TODO: Determine the exact format for this output.
+      let action = outcome;
+      if (fallbackReason) {
+        action = `${action}-${fallbackReason}`;
+      }
+      tracking.sendEvent({
+        action,
+        category: TAAR_IMPRESSION_CATEGORY,
+        label: newAddon.name,
+      });
+    }
+  }
+
+  cohort: CohortName;
+
+  dispatchFetchRecommendations({ guid, recommended }: Object) {
+    this.props.dispatch(fetchRecommendations({
+      errorHandlerId: this.props.errorHandler.id,
+      guid,
+      recommended,
+    }));
+  }
+
+  render() {
+    const { className, i18n, recommendations } = this.props;
+
+    if (!recommendations) {
+      log.debug(
+        'No recommandations, hiding the AddonRecommendations component.'
+      );
+      return null;
+    }
+
+    const { addons, loading, outcome } = recommendations;
+    const classnames = makeClassName('AddonRecommendations', className);
+
+    let header = loading ? <LoadingText width={100} /> : null;
+    if (!header) {
+      header = outcome === OUTCOME_RECOMMENDED ?
+        i18n.gettext('Other users with this extension also installed') :
+        i18n.gettext('Other popular extensions');
+    }
+
+    return (
+      <AddonsCard
+        addonInstallSource={outcome || ''}
```

Comment:
Is it ok to have empty string as install source?

Reply:
I believe so. Looking at the code that actually uses it, in [`SearchResult`](https://github.com/mozilla/addons-frontend/blob/master/src/amo/components/SearchResult/index.js#L144-L146), it simply checks `if (addonInstallSource)` and `''` is falsey, so it should yield a correct result.

Having said that, the only time the `outcome` will be missing is if we're rendering the loading version of the card, in which case there won't be anything to link anyway.